apps/docs/src/app/documentation/component-docs/action-bar/action-bar-docs.component.ts

```diff
@@ -1,41 +1,71 @@
-import { Component } from '@angular/core';
+import { Component, ViewChildren, ElementRef, QueryList, OnInit, AfterViewInit } from '@angular/core';
 
 import * as backButtonExample from '!raw-loader!./examples/action-bar-back-example.component.html';
 import * as titleTruncationExample from '!raw-loader!./examples/action-bar-long-string-title-truncation-example.component.html';
 import * as contextualMenuExample from '!raw-loader!./examples/action-bar-contextual-menu-example.component.html';
 import * as mobileExample from '!raw-loader!./examples/action-bar-mobile-example.component.html';
 import * as noBackButtonExample from '!raw-loader!./examples/action-bar-no-back-example.component.html';
 import { ExampleFile } from '../../core-helpers/code-example/example-file';
+import { DocsSectionTitleComponent } from '../../core-helpers/docs-section-title/docs-section-title.component';
+import { ActivatedRoute } from '@angular/router';
 
 @Component({
     selector: 'app-toolbar',
     templateUrl: './action-bar-docs.component.html',
     styleUrls: ['action-bar-docs.component.scss']
 })
-export class ActionBarDocsComponent {
-
-    backButtonExample: ExampleFile[] = [{
-        language: 'html',
-        code: backButtonExample
-    }];
-
-    titleTruncationExample: ExampleFile[] = [{
-        language: 'html',
-        code: titleTruncationExample
-    }]
-
-    noBackButtonExample: ExampleFile[] = [{
-        language: 'html',
-        code: noBackButtonExample
-    }];
-
-    actionsContextualMenuHtml: ExampleFile[] = [{
-        language: 'html',
-        code: contextualMenuExample
-    }];
-
-    mobileViewHtml: ExampleFile[] = [{
-        language: 'html',
-        code: mobileExample
-    }];
+export class ActionBarDocsComponent implements OnInit, AfterViewInit {
+    backButtonExample: ExampleFile[] = [
+        {
+            language: 'html',
+            code: backButtonExample
+        }
+    ];
+
+    titleTruncationExample: ExampleFile[] = [
+        {
+            language: 'html',
+            code: titleTruncationExample
+        }
+    ];
+
+    noBackButtonExample: ExampleFile[] = [
+        {
+            language: 'html',
+            code: noBackButtonExample
+        }
+    ];
+
+    actionsContextualMenuHtml: ExampleFile[] = [
+        {
+            language: 'html',
+            code: contextualMenuExample
+        }
+    ];
+
+    mobileViewHtml: ExampleFile[] = [
+        {
+            language: 'html',
+            code: mobileExample
+        }
+    ];
+    private fragment: any;
+    @ViewChildren(DocsSectionTitleComponent, { read: ElementRef }) myList: QueryList<ElementRef>;
+
+    constructor(private route: ActivatedRoute) {}
+
+    ngOnInit() {
+        this.route.fragment.subscribe(fragment => {
+            this.fragment = fragment;
+        });
+    }
+
+    ngAfterViewInit(): void {
```

Comment:
I would recommend putting the scroll functionality in `docs-section-title.component`

Reply:
Try also to avoid using `for` loop in the code, instead of it you can use `.forEach`, or `find / filter`.